src/glum/_glm.py

```diff
@@ -1873,8 +1863,70 @@ def _expand_categorical_penalties(penalty, X, drop_first):
         elif isinstance(X, np.ndarray):
             X = tm.DenseMatrix(X)
 
+        if not hasattr(self, "feature_names_") or not hasattr(self, "term_names_"):
+            self._get_feature_names(X)
+
         return X, y, sample_weight, offset, weights_sum, P1, P2
 
+    def _get_feature_names(self, X: ArrayLike):
```

Comment:
Could this method get a different name, please? I associated "get" with a low overhead getter (i.e. just return an attribute). This method doesn't return anything and it actually sets two fitted attributes. 

I would think it's a bit cleaner if this method just returned the feature names and terms and leave setting them to `_set_up_and_check_fit_args`.